src/utils/desktop.js

```diff
@@ -110,6 +111,21 @@ export const isVisible = w => w &&
   !w.getState('minimized') &&
   w.getState('focused');
 
+/**
+ * Allows for requesting whether a file exists without `async`
+ * @todo Move to VFS, so the 404 error doesn't log to console
+ * @param {string} urlToFile
+ * @returns {boolean}
+ */
+const syncRequest = (urlToFile) => {
```

Comment:
Synchronous requests blocks the main thread. So any time this function runs, the entire UI is frozen until the request finishes.

Also this use is deprecated in all browsers.

So there needs to be some alternate way of solving this. Ideally without any network request at all. The client should be aware of what's possible or not, and not do queries.

PS: It's possible to build OS.js in a standalone mode that does not require a server for use. And this would break those kinds of deployments as you can't do HTTP requests in a `file:///` environment.